File: module_me/src/main/java/com/fly/tour/me/NewsTypeAddActivity.java
package com.fly.tour.me;

import android.app.Activity;
import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProvider;
import android.content.Intent;
import android.support.annotation.Nullable;

import com.fly.tour.common.event.EventCode;
import com.fly.tour.common.event.me.NewsTypeCrudEvent;
import com.fly.tour.common.mvvm.BaseMvvmActivity;
import com.fly.tour.me.databinding.ActivityNewsTypeAddBinding;
import com.fly.tour.me.mvvm.factory.MeViewModelFactory;
import com.fly.tour.me.mvvm.viewmodel.NewsTypeAddViewModel;

import org.greenrobot.eventbus.EventBus;

/**
 * Description: <NewsTypeAddActivity><br>
 * Author:      mxdl<br>
<<<<<<< HEAD
 * Date:        2019/5/24<br>
=======
 * Date:        2019/07/02<br>
>>>>>>> 4.1.0
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public class NewsTypeAddActivity extends BaseMvvmActivity<ActivityNewsTypeAddBinding,NewsTypeAddViewModel> {

    @Override
    public int onBindLayout() {
        return R.layout.activity_news_type_add;
    }

    @Override
    public Class<NewsTypeAddViewModel> onBindViewModel() {
        return NewsTypeAddViewModel.class;
    }

    @Override
    public ViewModelProvider.Factory onBindViewModelFactory() {
        return MeViewModelFactory.getInstance(getApplication());
    }
    @Override
    public void initViewObservable() {
        mViewModel.getAddNewsTypeSuccViewEvent().observe(this, new Observer<Void>() {
            @Override
            public void onChanged(@Nullable Void aVoid) {
                EventBus.getDefault().post(new NewsTypeCrudEvent(EventCode.MeCode.NEWS_TYPE_ADD));
                setResult(Activity.RESULT_OK, new Intent());
                finishActivity();
            }
        });
    }

    @Override
    public int onBindVariableId() {
        return BR.viewModel;
    }

}

File: module_me/src/main/java/com/fly/tour/me/NewsTypeAddActivity.kt
package com.fly.tour.me

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import com.fly.tour.common.base.BaseMvpActivity
import com.fly.tour.common.event.EventCode
import com.fly.tour.common.event.me.NewsTypeCrudEvent
import com.fly.tour.common.util.ToastUtil
import com.fly.tour.me.contract.NewsTypeAddContract
import com.fly.tour.me.model.NewsTypeAddModel
import com.fly.tour.me.presenter.NewsTypeAddPresenter
import kotlinx.android.synthetic.main.activity_news_type_add.*
import org.greenrobot.eventbus.EventBus
/**
 * Description: <NewsTypeAddActivity><br>
 * Author:      mxdl<br>
 * Date:        2020/2/17<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
class NewsTypeAddActivity :
    BaseMvpActivity<NewsTypeAddModel, NewsTypeAddContract.View, NewsTypeAddPresenter>(),
    NewsTypeAddContract.View {
    override fun initPresenter(): NewsTypeAddPresenter {
        return NewsTypeAddPresenter(this)
    }

    override fun initListener() {
        mViewMeNewsType.enableEditContext(true)
        mBtnMeNewsType.setOnClickListener {
            var content = mViewMeNewsType.content
            if(TextUtils.isEmpty(content)){
                ToastUtil.showToast("请输入新闻类型")
                return@setOnClickListener
            }
            mPresenter?.addNewsType(content!!)
        }
    }

    override fun onBindLayout(): Int {
        return R.layout.activity_news_type_add
    }

    override fun initData() {
    }

    override fun finishActivity() {
        EventBus.getDefault().post(NewsTypeCrudEvent<Any>(EventCode.MeCode.NEWS_TYPE_ADD))
        setResult(Activity.RESULT_OK, Intent())
        super.finishActivity()
    }
}
